test_cluster_migration_while_seeding

https://github.com/dragonflydb/dragonfly/actions/runs/13198374590/job/36844631873#step:12:890


```
        capture = await seeder.capture_fake_redis()
        assert await seeder.compare(capture, instances[1].port)
    
        line = stop_and_get_restore_log(nodes[0].instance)
        assert extract_int_after_prefix("Keys skipped ", line) == 0
        assert extract_int_after_prefix("buckets skipped ", line) > 0
        assert extract_int_after_prefix("keys written ", line) >= 9_000
>       assert extract_int_after_prefix("buckets on_db_update ", line) > 0
E       AssertionError: assert 0 > 0
E        +  where 0 = extract_int_after_prefix('buckets on_db_update ', 'I20250207 11:07:42.041096 29736 streamer.cc:241] RestoreStreamer LSN of [0, 16383], shard 0 attempt 1 with 67485 commands. Buckets looped 897, buckets on_db_update 0, buckets skipped 1, buckets written 897. Keys skipped 0, keys written 9742\n')

dragonfly/cluster_test.py:2156: AssertionError

```

--- a/src/server/journal/streamer.cc
+++ b/src/server/journal/streamer.cc
@@ -21,6 +21,9 @@ ABSL_FLAG(uint32_t, replication_timeout, 30000,
 ABSL_FLAG(uint32_t, replication_stream_output_limit, 64_KB,
           "Time to wait for the replication output buffer go below the throttle limit");
 
+ABSL_FLAG(uint32_t, migration_buckets_serialization_threshold, 100,
+          "The Number of buckets to serialize on each iteration before yielding");
+
 namespace dfly {
 using namespace util;
 using namespace journal;
@@ -32,6 +35,7 @@ iovec IoVec(io::Bytes src) {
 }
 
 uint32_t replication_stream_output_limit_cached = 64_KB;
+uint32_t migration_buckets_serialization_threshold_cached = 100;
 
 }  // namespace
 
@@ -186,6 +190,8 @@ RestoreStreamer::RestoreStreamer(DbSlice* slice, cluster::SlotSet slots, journal
                                  ExecutionState* cntx)
     : JournalStreamer(journal, cntx), db_slice_(slice), my_slots_(std::move(slots)) {
   DCHECK(slice != nullptr);
+  migration_buckets_serialization_threshold_cached =
+      absl::GetFlag(FLAGS_migration_buckets_serialization_threshold);
   db_array_ = slice->databases();  // Inc ref to make sure DB isn't deleted while we use it
 }
 
@@ -229,7 +235,7 @@ void RestoreStreamer::Run() {
       stats_.buckets_loop += WriteBucket(it);
     });
 
-    if (++last_yield >= 100) {
+    if (++last_yield >= migration_buckets_serialization_threshold_cached) {
       ThisFiber::Yield();
       last_yield = 0;
     }

--- a/tests/dragonfly/cluster_test.py
+++ b/tests/dragonfly/cluster_test.py
@@ -2136,7 +2136,9 @@ async def test_cluster_migration_huge_container(df_factory: DflyInstanceFactory)
     assert extract_int_after_prefix("buckets on_db_update ", line) == 0
 
 
-@dfly_args({"proactor_threads": 2, "cluster_mode": "yes"})
+@dfly_args(
+    {"proactor_threads": 2, "cluster_mode": "yes", "migration_buckets_serialization_threshold": 3}
+)
 @pytest.mark.parametrize("chunk_size", [1_000_000, 30])
 @pytest.mark.asyncio
 @pytest.mark.exclude_epoll

